yash-env/src/system/real.rs

```diff
@@ -670,20 +671,31 @@
     /// process ID. In the child, the starter runs the task and exits the
     /// process.
     fn new_child_process(&mut self) -> Result<ChildProcessStarter> {
-        use nix::unistd::ForkResult::*;
-        // SAFETY: As stated on RealSystem::new, the caller is responsible for
-        // making only one instance of RealSystem in the process.
-        match unsafe { nix::unistd::fork()? } {
-            Parent { child } => Ok(Box::new(move |_env, _task| {
-                Box::pin(std::future::ready(Pid::from_nix(child)))
-            })),
-            Child => Ok(Box::new(|env, task| {
-                Box::pin(async move {
-                    task(env).await;
-                    std::process::exit(env.exit_status.0)
-                })
-            })),
+        let raw_pid = unsafe { nix::libc::fork() }.errno_if_m1()?;
```

Comment:
`nix::unistd::fork()` is also an unsafe API, so we need to handle the unsafe blocks anyway.

We're now trying to minimize the dependency on nix because it is becoming too high-level for us to use as a low-level API.